dotCMS/src/main/java/com/dotcms/publishing/listener/PushPublishKeyResetEventListener.java

```diff
@@ -0,0 +1,63 @@
+package com.dotcms.publishing.listener;
+
+import com.dotcms.publisher.endpoint.bean.PublishingEndPoint;
+import com.dotcms.publisher.endpoint.business.PublishingEndPointAPI;
+import com.dotcms.system.event.local.model.EventSubscriber;
+import com.dotcms.system.event.local.type.security.CompanyKeyResetEvent;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Logger;
+import com.liferay.util.Encryptor;
+import java.security.Key;
+import java.util.List;
+
+/**
+ * Once registered this listener takes care or recreating the Push-Publish Auth-Keys
+ * First decrypting the old content using the original Key then re-inserting it using the new Key.
+ */
+public class PushPublishKeyResetEventListener implements EventSubscriber<CompanyKeyResetEvent> {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A class which only has private constructors should be final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5677177)